src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -672,23 +682,30 @@ event::ptr primitive_inst::realloc_if_needed() {
         const auto& ibuf_layouts = _impl->get_internal_buffer_layouts();
         if (ibuf_layouts.empty())
             return ev;
-
         for (size_t i = 0; i < ibuf_layouts.size(); ++i) {
             if (i < _intermediates_memory.size() && ibuf_layouts[i].bytes_count() <= max_intermediates_memory_sizes[i]) {
                 // can reuse
                 _intermediates_memory[i] = _network.get_engine().reinterpret_buffer(*_intermediates_memory[i], ibuf_layouts[i]);
+                std::string mem_info = "_ibuf" + std::to_string(i) + ":new_alloc";
+                GPU_DEBUG_PROFILED_STAGE_MEMALLOC_INFO(mem_info);
             } else {
                 // TODO: If there is a kernel which requires reset internal buffer in the future,
                 // we'll need additional handle for that purpose like need_reset_output_memory
                 bool need_reset = false;
+                bool from_pool = false;
                 if (i < _intermediates_memory.size()) {
-                    _intermediates_memory[i] = allocate_internal_buffer(i, need_reset);
+                    _intermediates_memory[i] = allocate_internal_buffer(i, need_reset, &from_pool);
                     max_intermediates_memory_sizes[i] = _intermediates_memory[i]->size();
                 } else {
                     // i-th layout has not been allocated yet
-                    _intermediates_memory.push_back(allocate_internal_buffer(i, need_reset));
+                    _intermediates_memory.push_back(allocate_internal_buffer(i, need_reset, &from_pool));
                     max_intermediates_memory_sizes.push_back(_intermediates_memory[i]->size());
                 }
+                if (from_pool) {
+                    std::string mem_info = "_ibuf" + std::to_string(i) + ":from_pool";
+                } else {
+                    std::string mem_info = "_ibuf" + std::to_string(i) + ":from_pool";
```

Comment:
It looks like `:new_alloc` was intended here

Reply:
Fixed! thank you for checking!